src/library/scala/annotation/meta/constructor.scala

```diff
@@ -0,0 +1,19 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___     Scala API                            **
+**    / __/ __// _ | / /  / _ |    (c) 2002-2013, LAMP/EPFL             **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/               **
+** /____/\___/_/ |_/____/_/ | |                                         **
+**                          |/                                          **
+\*                                                                      */
+package scala.annotation.meta
+
+/**
+ * An annotation marked with `@constructor` signals that an annotation can be
+ * placed at a constructor.
```

Comment:
Mhh, looks I got some only slightly related changes in this PR. I think I'll remove those things for now, as the implementation is not ready.

Thanks for catching it!